dashboard/app/models/user.rb

```diff
@@ -330,6 +330,7 @@ def hash_email
   def hide_email_and_full_address_for_students
     if student?
       self.email = ''
+      self.unconfirmed_email = nil
```

Comment:
Because...  It shouldn't be. But we require `email` to be non-NULL.

`email varchar(255) COLLATE utf8_unicode_ci NOT NULL DEFAULT '',`

but we don't make that requirement on `unconfirmed_email`.